content/tutorials/custom-zk-chain/20.customizing-your-chain.md

```diff
@@ -0,0 +1,393 @@
+---
+title: Customizing your ZK chain
+description: Create a ZK chain with a custom ERC20 base token.
+
+---
+
+One way you can customize your ZK chain is by changing the base token used for gas from ETH to an ERC20 token.
+Let's try customizing a new chain by using a custom ERC20 base token.
+
+With the initial release of ZK stack, custom ERC20 base tokens must be added to an allowlist for use as a base token for a chain.
+The allowed addresses are stored and checked in the `BridgeHub` contract on the L1.
+In the future it will be possible to add a new token in a permissionless process.
+
+For now, you have the ability to add any tokens to the allowlist in your local ecosystem.
+
+The overall flow for setting up a local chain with a custom ERC20 base token looks like this:
+
+1. Deploy an ERC20 token to the L1.
+1. Create your new chain that uses the ERC20 token and set it as the default.
+1. Send ERC20 tokens to the ecosystem and chain governor addresses on the L1.
+1. Initialize the new chain in the ecosystem.
+1. Bridge tokens from the L1 to your new chain.
```

Comment:
Where in this process is the ERC20 token added in the allowlist on the BridgeHub contract? It looks like from the rest of the guide that we ourselves aren't adding it to to the allowlist, so is it a part of the initialization of a ZK Chain?

Reply:
you don't have to do it directly, it happens in the background when you initialize the chain. The part at the top is there mainly to say that although we can do all of this in a tutorial, it's not a permissionless process to do it on testnet or mainnet

Reply:
OH now I get it. It feels like there's a lot being indirectly addressed in the description above. So in production, you have to go through a process to get a token added to an allowlist to be used as a base token, but with the local installation we automate this allowlisting process for development ease, is that correct?